pkg/restore/actions/pod_volume_restore_action.go

```diff
@@ -60,10 +62,34 @@ type PodVolumeRestoreAction struct {
 }
 
 func NewPodVolumeRestoreAction(logger logrus.FieldLogger, client corev1client.ConfigMapInterface, crClient ctrlclient.Client, namespace string) (*PodVolumeRestoreAction, error) {
-	deployment := &appsv1api.Deployment{}
-	if err := crClient.Get(context.TODO(), types.NamespacedName{Name: "velero", Namespace: namespace}, deployment); err != nil {
-		return nil, err
+	var (
+		ctx        = context.TODO()
+		deployment = new(appsv1api.Deployment)
+	)
+	if err := crClient.Get(ctx, types.NamespacedName{Name: "velero", Namespace: namespace}, deployment); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return nil, err
+		}
+
+		// fallback to avoid #9023
+		var (
+			deployments = new(appsv1api.DeploymentList)
+			ls          = labels.SelectorFromValidatedSet(install.Labels())
+		)
+		if err := crClient.List(ctx, deployments, &ctrlclient.ListOptions{
+			LabelSelector: ls,
+			Limit:         1,
+		}); err != nil {
+			return nil, err
+		}
+
+		if len(deployments.Items) == 0 {
+			return nil, fmt.Errorf("could not find velero deployment by the name nor by a label selector %q", ls)
+		}
+
+		deployment = &deployments.Items[0]
```

Comment:
in standard Velero installations, the code:

```Go
deployment = &deployments.Items[0]
```
should reliably select the Velero server Deployment, not a Node Agent, which is a daemonset.

Reply:
Resolving because of a new unified function